src/core/lib/gpr/stat.h

```diff
@@ -0,0 +1,31 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_CORE_LIB_GPR_STAT_H
+#define GRPC_CORE_LIB_GPR_STAT_H
+
+#include <grpc/support/port_platform.h>
+
+#include <stdio.h>
+#include <time.h>
+
+/* Gets the last-modified timestamp of a file or a directory. If error occurs,
+ * the epoch time(int value 0) will be returned. */
```

Comment:
That doesn't seem like a good API for returning the error.  What if a file actually has the epoch as its timestamp?

I suggest making this a C++ API in the ../gprpp directory.  Then we can do something like this:

```
absl::Status GetFileModifyTimestamp(absl::string_view filename, time_t* timestamp);
```

(Ideally, we would use `absl::StatusOr<>` here, but I don't think we can use that yet.)